File: scratch/check_syntax.js

const fs = require('fs');
const content = fs.readFileSync('c:\\Users\\USER\\Desktop\\MCMS\\main.js', 'utf8');

function check(charOpen, charClose, name) {
    let open = 0;
    let close = 0;
    for (let i = 0; i < content.length; i++) {
        if (content[i] === charOpen) open++;
        if (content[i] === charClose) close++;
    }
    console.log(`${name}: open=${open}, close=${close}`);
}

check('{', '}', 'Braces');
check('(', ')', 'Parens');
check('[', ']', 'Brackets');

File: scratch/check_syntax.py

import sys

def count_tokens(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    backticks = content.count('`')
    braces_open = content.count('{')
    braces_close = content.count('}')
    parens_open = content.count('(')
    parens_close = content.count(')')
    
    print(f"Backticks: {backticks} (Odd means error)")
    print(f"Braces: {braces_open} open, {braces_close} close")
    print(f"Parens: {parens_open} open, {parens_close} close")

if __name__ == "__main__":
    count_tokens(sys.argv[1])
